jenkins-dsl/cassandra_pipeline.groovy

```diff
@@ -305,6 +305,17 @@ pipeline {
           }
         }
       }
+      stage('Dependency check') {
+          steps {
+              sh "ant -f ./cassandra/build.xml dependency-check"
+              sh "xz ./cassandra/build/dependency-check-report.html"
+          }
+          post {
+              always {
+                  sshPublisher(publishers: [sshPublisherDesc(configName: 'Nightlies',transfers: [sshTransfer(remoteDirectory: 'cassandra/devbranch/${JOB_NAME}/${BUILD_NUMBER}/', sourceFiles: './cassandra/build/dependency-check-report.html.xz')])])
```

Comment:
Is there any point in compressing and uploading the `dependency-check-report.html` if we hard fail the CI run if there's any violations?

Just failing, and letting the developer figure it out, seems the KISS solution (as everything is encapsulated to cassandra-artifacts.sh and it is easy and quick for the developer to run dependency-check).